### DISCUSSION TITLE
VPN client connect to Checkpoint with 2FA (token otp get from FortiToken)

### DISCUSSION THREAD

**Discussion by nguyentthai96** [strongswan/strongswan#2268] [Q&A]
Hi, I had an issue relate connect to Checkpoint, I researched can using strongswan the best to connect.

Before, I used SNX to connect the client from my PC to Checkpoint on Ubuntu Linux, but now server check upgrades require 2FA MFA authen with OTP get from app [FortiToken](https://apps.apple.com/us/app/fortitoken-mobile/id500007723) and user name password
https://sc1.checkpoint.com/documents/SSL_Network_Extender_AdminGuide/Content/Topics-SNX-Admin-Guide/Getting-Started.htm   

Now, I try on windows connect with Checkpoint tools client is working normally, but I no solution on Linux Ubuntu
(On Windows open Checkpoint tools client -> input username/password -> connecting to server response -> enter OTP get FortiToken ==> connected normally)

I researched, and some members recommend using the best  strongswan, but I cannot find setup tools or any GUI tutorial connect  terminal/guideline.

https://sc1.checkpoint.com/documents/R81.10/WebAdminGuides/EN/CP_R81.10_RemoteAccessVPN_AdminGuide/Topics-VPNRG/strongSwan-Client-Support.htm
https://community.checkpoint.com/t5/Remote-Access-VPN/strongSwan-GUI-Network-Manager-Username-Password/td-p/182449

https://community.checkpoint.com/t5/Remote-Access-VPN/Endpoint-VPN-MFA-client-for-Linux/m-p/146910#M6952

**Comment by tobiasbrunner** [comment]
I doubt this will work. They probably use some proprietary extension to implement this (or even some out-of-band channel). The second to last page you linked even lists 2FA as a known limitation when using strongSwan.

**Comment by nguyentthai96** [comment]
My team use Checkpoint Endpoint VPN client for Windows to access a company VPN, using username/password and then confirmation on the phone using  FortiToken mobile authenticator. 
Is there any way to setup a client in this scenario for Linux (Ubuntu)?

Before, I use snx, but it looked like snx not support 2FA/MFA OTP token.
snx -s IP -u USER

I researched some recommend use strongswan the best but I cannot find any command line or GUI line tools tutorial use.
If had experience can share me some CLI or GUI install and use (I try install but similar not work) 

VPN server site use Checkpoint version R81.20
https://support.checkpoint.com/results/sk/sk165014

**Comment by nguyentthai96** [comment]
https://support.checkpoint.com/results/sk/sk171556

sudo nano /etc/ipsec.conf

`
config setup
    charondebug="ike 2, knl 2, cfg 2"

conn %default
    keyexchange=ikev2
    ike=aes256-sha256-modp2048!
    esp=aes256-sha256!

conn myvpn
    keyexchange=ikev2
    leftauth=eap-mschapv2
    eap_identity=ACCOUNT_VPN
    right=IP_SERVER_VPN
    rightid=%any
    rightauth=pubkey
    rightsubnet=0.0.0.0/0
    rightdns=8.8.8.8,8.8.4.4
    leftsourceip=%config
    auto=start
`


vpn_strongswan.sh

`
#!/bin/bash
echo ": EAP \"$1\"" > /etc/ipsec.secrets
sudo ipsec restart
sudo ipsec up myvpn
`

I search google and try here, 
but I'm confuse with Checkpoint client on Windows, 
I will enter username/password first and request to server VPN, when the response is success, continue enter OTP FortiToken mobile. 

About in my script enters password+OTP

**Comment by nguyentthai96** [comment]
With Checkpoint remote client on the windows 
After enter username/password -> then enter OTP from FortiToken here 
![imgpsh_fullsize_anim (1)](https://github.com/strongswan/strongswan/assets/14333403/1e84a0ba-f736-445c-94ca-0189659eba98)
